src/dashboard/dashboardgeneral.js

```diff
@@ -2,50 +2,118 @@ define(["jQuery", "loading", "fnchecked", "emby-checkbox", "emby-textarea", "emb
-                        Dashboard.processServerConfigurationUpdateResult(), requiresReload && !AppInfo.isNativeApp && window.location.reload(!0)
-                    })
+                    brandingConfig.LoginDisclaimer = form.querySelector("#txtLoginDisclaimer").value;
+                    brandingConfig.CustomCss = form.querySelector("#txtCustomCss").value;
+                    currentBrandingOptions && brandingConfig.CustomCss !== currentBrandingOptions.CustomCss && (requiresReload = !0);
+                    ApiClient.updateNamedConfiguration(brandingConfigKey, brandingConfig).then(function() {
+                        Dashboard.processServerConfigurationUpdateResult();
+                        if (requiresReload && !AppInfo.isNativeApp) {
+                            window.location.reload(true);
+                        }
+                    });
                 })
             })
         }), !1
     }
-    var currentBrandingOptions, currentLanguage, brandingConfigKey = "branding";
+
+    var currentBrandingOptions;
+    var currentLanguage;
+    var brandingConfigKey = "branding";
+
     return function(view, params) {
         $("#btnSelectCachePath", view).on("click.selectDirectory", function() {
             require(["directorybrowser"], function(directoryBrowser) {
                 var picker = new directoryBrowser;
                 picker.show({
                     callback: function(path) {
-                        path && (view.querySelector("#txtCachePath").value = path), picker.close()
+                        path && (view.querySelector("#txtCachePath").value = path);
+                        picker.close();
                     },
-                    validateWriteable: !0,
+                    validateWriteable: true,
                     header: Globalize.translate("HeaderSelectServerCachePath"),
                     instruction: Globalize.translate("HeaderSelectServerCachePathHelp")
                 })
             })
-        }), $(".dashboardGeneralForm", view).off("submit", onSubmit).on("submit", onSubmit), view.addEventListener("viewshow", function() {
-            var promise1 = ApiClient.getServerConfiguration(),
-                promise2 = ApiClient.getJSON(ApiClient.getUrl("Localization/Options")),
-                promise3 = ApiClient.getSystemInfo();
-            Promise.all([promise1, promise2, promise3]).then(function(responses) {
-                loadPage(view, responses[0], responses[1], responses[2])
-            }), ApiClient.getNamedConfiguration(brandingConfigKey).then(function(config) {
-                currentBrandingOptions = config, view.querySelector("#txtLoginDisclaimer").value = config.LoginDisclaimer || "", view.querySelector("#txtCustomCss").value = config.CustomCss || ""
+        });
+
+        $("#btnSelectMetadataPath", view).on("click.selectDirectory", function() {
```

Comment:
these two functions seem identical except what field they modify and what help text show, can copy-pasteness be reduced here?

Reply:
I would rather leave that alone for this batch of changes, I am just moving things around for now.